R/aggregate_age.R

```diff
@@ -0,0 +1,104 @@
+#' Aggregate age-specific counts to aggregate age groups
+#'
+#' @param dt data.table with data to be aggregated.
+#' * must only contain columns specified in `id_cols` and `value_cols`.
+#' * must include 'age_start' and 'age_end' columns.
+#' * each combination of `id_cols` must uniquely identify each row.
+#' @param id_cols character vector of id columns that uniquely identify each row
+#' of `dt`.
+#' * must include 'age_start' and 'age_end'.
+#' @param value_cols character vector of value columns to be aggregated.
+#' @param target_ages_dt data.table defining age groups to aggregate to.
+#' * must include 'age_start' and 'age_end' columns.
+#'
+#' @return data.table with `id_cols` and `value_cols` columns for aggregate age
+#'   groups.
+#'
+#' @export
+#'
+#' @examples
+#' input_dt <- data.table::data.table(year = 2010,
+#'                                    age_start = seq(0, 95, 1),
+#'                                    age_end = c(seq(1, 95, 1), 125),
+#'                                    value1 = 1, value2 = 2)
+#' target_dt <- data.table::data.table(age_start = c(0, 15, 85),
+#'                                     age_end = c(5, 60, 125))
+#' output_dt <- aggregate_age(input_dt,
+#'                            id_cols = c("year", "age_start", "age_end"),
+#'                            value_cols = c("value1", "value2"),
+#'                            target_ages_dt = target_dt)
+#'
+aggregate_age <- function(dt, id_cols, value_cols, target_ages_dt) {
+
+  # check `id_cols` argument
+  assertive::assert_is_character(id_cols)
+  assertthat::assert_that("age_start" %in% id_cols & "age_end" %in% id_cols,
+                          msg = "`id_cols` must include 'sex'.")
+
+  # check `value_cols` argument
+  assertive::assert_is_character(value_cols)
+
+  # check `dt` argument
+  assertive::assert_is_data.table(dt)
+  for (value_col in value_cols) {
+    assertive::assert_is_numeric(dt[[value_col]])
+  }
+  capture.output(assertable::assert_colnames(dt, c(id_cols, value_cols), only_colnames = T))
```

Comment:
Are you aware of the `quiet = T` option in assertable functions?

Reply:
Fixed!